zulipterminal/helper.py

```diff
@@ -92,29 +92,30 @@ def _set_count_in_model(new_count: int, changed_messages: List[Message],
         but updates `unread_counts` (which can update the model
         if it's passed in, but is not tied to it).
     """
+    # broader unread counts (for all_* and streams) are updated
+    # later conditionally.
+    def update_unreads(unreads: Any, key: Any) -> None:
+        if key in unreads:
+            unreads[key] += new_count
+            if unreads[key] == 0:
+                unreads.pop(key)
+        elif new_count == 1:
+            unreads[key] = new_count
+
     for message in changed_messages:
         if message['type'] == 'stream':
             key = (message['stream_id'], message['subject'])
-            unreads = unread_counts['unread_topics']
+            update_unreads(unread_counts['unread_topics'], key)
```

Comment:
Perhaps it would be clearer here to also put the `key`s inline, so we don't then need the key variables, including the extra variable you add in the second commit.